### ISSUE TITLE
Lack of Space Crash

### ISSUE THREAD

**Issue by shinra358** [nagadomi/nunif#267]
I spent 2 days trying to convert a movie to be met with a crash that said i was out of space. I would like a feature that pauses the routine and prompts you to free some space on your own so that you can come back and resume the project.

**Comment by nagadomi** [comment]
I can add this feature, but I feel it is a very generic OS issue that is not really related to iw3. Because all running processes affect the disk space.

If I were to implement this, it would be as follows,
1. Measure free disk space every few seconds
2. Predicts the number of seconds until the disk is full from multiple samples
3. If the disk is expected to become full in the near future (e.g., 1 hour), the software will pause on its own.

This may be a very annoying feature for users who are working with less disk space.
Basically, I think disk space should be managed by the user, not each individual software.

**Comment by stsnewyork83** [comment]
I have also encountered this problem. Is there a way for when it has an error, for the app to at least encapsulate and complete the file? because every time an error comes it leaves me with a broken .mp4 file and its difficult to repair afterwards.

**Comment by nagadomi** [comment]
1. Split the source video into multiple segments
2. Convert each segment video to 3D (If no space error occurs during conversion, it can be resumed from the previous segment)
3. Merge the results into one video

**Comment by nagadomi** [comment]
Is it possible to resume encoding when ffmpeg crashes during encoding and the mp4 is corrupted?
The resume issue is the same as that.
One possible workaround is to split the video into smaller pieces as described above.
